src/Workspaces/Core/Portable/GeneratedCodeRecognition/AbstractGeneratedCodeRecognitionService.cs

```diff
@@ -12,6 +12,16 @@ internal abstract class AbstractGeneratedCodeRecognitionService : IGeneratedCode
         public bool IsGeneratedCode(Document document, CancellationToken cancellationToken)
         {
             var syntaxTree = document.GetSyntaxTreeSynchronously(cancellationToken);
+
+            // First check if user has configured "generated_code = true | false" in .editorconfig
+            var analyzerOptions = document.Project.AnalyzerOptions.AnalyzerConfigOptionsProvider.GetOptions(syntaxTree);
```

Comment:
We already have https://github.com/dotnet/roslyn/blob/0c056f00034b54f4463160bd9e7afd59785e2c4a/src/Workspaces/Core/Portable/Workspace/Solution/Document.cs#L520 but that's async. I wonder if it's worth us adding a sync version of that too...

Reply:
I also first thought of using that API and even investigated making IsGeneratedCode async. However, that seems API seems to return options applicable for project's file path, not for the document file path (which does not seem intuitive honestly). Any case, I think this seemed more clear enough to me and wasn't too complex code to add it here.